### DISCUSSION TITLE
Thoughts on: benchmarking-top-open-source-speech-models blog

### DISCUSSION THREAD

**Discussion by desh2608** [deepgram/community#335] [General help] [answered]
https://deepgram.com/learn/benchmarking-top-open-source-speech-models

I have no idea what this comparison is intended to show. You are comparing a "framework" vs. an "encoder" vs. a "model" (talk about apples to oranges!).

* The Kaldi model is a hybrid HMM-DNN intended for composability (e.g. adding your own LM). It is trained on 10k hours.
* The wav2vec 2.0 is an encoder designed to be robust across tasks other than ASR (e.g. speaker ID). It is pretrained on 60k hours of LibriLight and fine-tuned on LibriSpeech with a CTC head.
* Whisper is an attention-based encoder-decoder trained on 680k hours of speech and specifically designed to work well for general-purpose ASR.

Reminds me of the saying: "if you judge a fish by its ability to climb a tree..."! The comparison leaves one wondering:

* What if you used the wav2vec 2.0 encoder in the Kaldi model? Would it then perform similar/better than the CTC one?
* What if you trained the wav2vec 2.0 encoder on 680k hours before fine-tuning? Would it perform similar to Whisper?
* How can we get composability with a strong performance? Perhaps one could take the Whisper encoder and plug it into a Kaldi-style WFST decoder.

(In fact, ESPNet and NeMo have several models which achieve similar performance as Whisper while being "transparent".)

**Comment by hbredin** [comment]
Talking about apples and oranges, [that one](https://twitter.com/DeepgramAI/status/1656710373217816576) is "fruity" too ;-)

**Comment by scottstephenson** [comment, marked as answer]
Great feedback. I think a good way to think about the premise of that blog post is: "X when used to do ASR" or " X when used to do ASR in the way that most folks try to use it". I.e. the vast majority of ASR users are not speech recognition experts and cannot decompose and recompose different parts of models to get an end result they would like (audio converted to text). Instead they will take whatever is as close to what they need as they can get, wrap it in a server and hope it does a good job for them. This is true of kaldi, wav2vec, whisper, espnet, nemo, and more. This is a very reasonable way to operate if you are an engineer just trying to solve your problem. Of course it has downsides like: a) you have to pay to set up your infra, b) you have to find a way to update the models to keep up with the jones', c) you are responsible for making the service have a good SLA for yourself, etc.

On the other hand, if you are a researcher and are willing to do the work to recompose the models and then find a way to wrap it in a robust engineering framework, that's another story of course. This is not what the vast majority of ASR users are doing though and that blog post is intended for them. I can see how the blog post would be annoying for someone who does know all of the nuances though!

**Reply by desh2608** [reply]
Thanks for your reply, Scott. I agree that for a common user, ease-of-use and robustness of transcription results are most important, and that is probably where Whisper shines :)
